panic: runtime error: index out of range listing nodes

panic: runtime error: index out of range listing nodes in Dashboard

```
Dashboard version: 1.6.2
Kubernetes version: 1.5.2
Operating system: RHEL 7.3
Docker version: 1.12.6
Heapster version: 1.4.0
```

##### Steps to reproduce
Open "Nodes" option in dashboard

##### Observed result
Panic

STDERR:

Getting lists of all workloads
Getting pod metrics
[2017-07-26T15:30:54Z] Outcoming response to 10.10.15.1:41004 with 200 status code
[2017-07-26T15:31:01Z] Incoming HTTP/1.1 GET /api/v1/node?filterBy=&itemsPerPage=15&namespace=&page=1&sortBy=d,creationTimestamp request from 10.10.15.1:41004: {}
panic: runtime error: index out of range

goroutine 151 [running]:
panic(0x1c6d8e0, 0xc420014060)
	/home/travis/.gimme/versions/go1.7.1.linux.amd64/src/runtime/panic.go:500 +0x1a1
github.com/kubernetes/dashboard/src/app/backend/integration/metric/heapster.heapsterClient.ithResourceDownload.func1(0x2cc5e60, 0xc42047b1d0, 0x1e95537, 0x4, 0x1e97a5e, 0x6, 0xc4208bbc00, 0x15, 0x20, 0xc420363710, ...)
	/home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/integration/metric/heapster/client.go:171 +0x5a5
created by github.com/kubernetes/dashboard/src/app/backend/integration/metric/heapster.heapsterClient.ithResourceDownload
	/home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/integration/metric/heapster/client.go:177 +0x109

--- a/src/app/backend/integration/metric/heapster/client.go
+++ b/src/app/backend/integration/metric/heapster/client.go
@@ -162,16 +162,21 @@ func (self heapsterClient) ithResourceDownload(selector heapsterSelector, metric
 		}
 		dataPoints := DataPointsFromMetricJSONFormat(rawResult)
 
-		result.Metric <- &metricapi.Metric{
+		metric := &metricapi.Metric{
 			DataPoints:   dataPoints,
 			MetricPoints: toMetricPoints(rawResult.Metrics),
 			MetricName:   metricName,
-			Label: metricapi.Label{
+		}
+
+		if len(selector.Label[selector.TargetResourceType]) > i {
+			metric.Label = metricapi.Label{
 				selector.TargetResourceType: []types.UID{
 					selector.Label[selector.TargetResourceType][i],
 				},
-			},
+			}
 		}
+
+		result.Metric <- metric
 		result.Error <- nil
 		return
 	}()
@@ -200,16 +205,21 @@ func (self heapsterClient) allInOneDownload(selector heapsterSelector, metricNam
 		for i, rawResult := range rawResults.Items {
 			dataPoints := DataPointsFromMetricJSONFormat(rawResult)
 
-			result[i].Metric <- &metricapi.Metric{
+			metric := &metricapi.Metric{
 				DataPoints:   dataPoints,
 				MetricPoints: toMetricPoints(rawResult.Metrics),
 				MetricName:   metricName,
-				Label: metricapi.Label{
+			}
+
+			if len(selector.Label[selector.TargetResourceType]) > i {
+				metric.Label = metricapi.Label{
 					selector.TargetResourceType: []types.UID{
 						selector.Label[selector.TargetResourceType][i],
 					},
-				},
+				}
 			}
+
+			result[i].Metric <- metric
 			result[i].Error <- nil
 		}
 		return

